x-pack/solutions/security/plugins/security_solution/public/reports/hooks/use_download_ai_value_report.tsx

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { AI_VALUE_REPORT_LOCATOR } from '@kbn/deeplinks-analytics';
+import { useMemo } from 'react';
+import { useKibana } from '../../common/lib/kibana';
+import type { AIValueReportParams } from '../../../common/locators/ai_value_report/locator';
+import { useAIValueExportContext } from '../providers/ai_value/export_provider';
+
+interface UseDownloadAIValueReportParams {
+  anchorElement: HTMLElement | null;
+  timeRange: AIValueReportParams['timeRange'];
+}
+
+export const useDownloadAIValueReport = ({
+  anchorElement,
+  timeRange,
+}: UseDownloadAIValueReportParams) => {
+  const { share: shareService, serverless } = useKibana().services;
+  const isServerless = !!serverless;
+  const aiValueExportContext = useAIValueExportContext();
+
+  const forwardedState = useMemo(() => {
+    if (!aiValueExportContext?.buildForwardedState) {
+      return undefined;
+    }
+
+    return aiValueExportContext.buildForwardedState({ timeRange });
+  }, [timeRange, aiValueExportContext]);
```

Comment:
```suggestion
  }, [timeRange, aiValueExportContext?.buildForwardedState]);
```

Less rerenders if you depend on `aiValueExportContext?.buildForwardedState` directly